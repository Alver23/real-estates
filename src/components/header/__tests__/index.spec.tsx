// Dependencies
import React, { Dispatch } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

// Under test file
import Header from '../index';

describe('<Header />', () => {
  let component: ShallowWrapper;
  const mockState: [boolean, Dispatch<any>] = [true, jest.fn()];

  beforeEach(() => {
    jest.spyOn(React, 'useState').mockReturnValue(mockState);
    component = shallow(<Header />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the click event of the hamburger menu', () => {
    component.find('button').simulate('click');
    expect(mockState[1]).toHaveBeenCalledTimes(1);
  });
});
