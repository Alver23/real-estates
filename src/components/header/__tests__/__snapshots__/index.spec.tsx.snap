// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should save a snapshot of the component 1`] = `
<header
  className="flex flex-row justify-between items-center md:px-20 w-full h-14 | border-b border-green-100"
>
  <div
    className="flex flex-row items-end"
  >
    <button
      className="relative md:hidden px-4"
      onClick={[Function]}
      type="button"
    >
      <VscMenu
        className="text-green-400"
        size={30}
      />
      <div
        className="absolute top-0 right-2"
      >
        <NotificationBar />
      </div>
    </button>
    <Link
      className="block"
      to="/"
    >
      <img
        alt="La Haus"
        className="hidden md:block text-green-800"
        src="logo-desktop.svg"
        title="La Haus Logo"
      />
      <img
        alt="La Haus"
        className="md:hidden text-green-800"
        src="logo.svg"
        title="La Haus Logo"
      />
    </Link>
  </div>
  <div
    className="hidden md:block"
  >
    <Navbar />
  </div>
  <div
    className="md:hidden absolute w-full top-14"
  >
    <Navbar />
  </div>
</header>
`;
